projects/core/src/config/utils/dynamic-template.ts

```diff
@@ -1,10 +1,24 @@
 export class DynamicTemplate {
-  static resolve(templateString: string, templateVariables: Object) {
+  /**
+   * Populates the given template with the variables provided
+   *
+   * @param templateString template of the OCC endpoint
+   * @param templateVariables variables to replace in the template
+   * @param encodeVariable encode variable before placing it in the template
+   */
+  static resolve(
+    templateString: string,
+    templateVariables: Object,
+    encodeVariable?: boolean
+  ): string {
     for (const variableLabel of Object.keys(templateVariables)) {
       const placeholder = new RegExp('\\${' + variableLabel + '}', 'g');
       templateString = templateString.replace(
         placeholder,
-        templateVariables[variableLabel]
+        // TODO 4.0: default to encodeVariable = true
+        encodeVariable
+          ? encodeURIComponent(templateVariables[variableLabel])
```

Comment:
Should we also apply this change in CDS `DynamicTemplate`?

Reply:
I noticed CDS has their own emplementation of the `occ-endpoints.service`. I'd want to clarify why before changing anything in CDS. I would like to move this to a separate issue.

Reply:
Ok. Let's create a ticket for that.